terracotta/client/app/src/sidebar/VerticalHandle/VerticalHandle.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect, useCallback, FC } from 'react'
+import Handle from './Handle'
+
+const minWidth = 0
+const minMapWidth = 50
+
+interface Props{
+	boxWidth: number,
+	onDrag: (w: number) => void,
+}
+
+const VerticalHandle: FC<Props> = ({
+	boxWidth,
+	onDrag,
+}) => {
+
+	const [ isDragging, setIsDragging ] = useState(false)
+	const [ initialWidth, setInitialWidth ] = useState(minWidth * 2)
+
+	const handleMouseMove = useCallback(e => {
+
+		const windowWidth = window.innerWidth
+		const w = e.clientX < minMapWidth ?
+			windowWidth - minMapWidth :
+			windowWidth - e.clientX
+
+		onDrag(w)
+
+	}, [ onDrag ])
+
+	const handleMouseDown = useCallback(e => {
+
+		setIsDragging(true)
+		const w = window.innerWidth - e.clientX
+		setInitialWidth(initialWidth < minWidth ? minWidth : w)
+
+	}, [  initialWidth ])
+
+	const handleMouseUp = useCallback(() => {
+
+		if (boxWidth < minWidth) {
+
+			onDrag(initialWidth)
+
+		}
```

Comment:
This can be removed?

Reply:
For our case it's indeed not needed since it's `0`, but for reusability reasons, I would keep it. I'll instead pass it as optional props to that component.